agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,277 @@
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.supercsv.cellprocessor.Optional;
+import org.supercsv.cellprocessor.constraint.NotNull;
+import org.supercsv.cellprocessor.ift.CellProcessor;
+import org.supercsv.io.CsvBeanReader;
+import org.supercsv.io.ICsvBeanReader;
+import org.supercsv.prefs.CsvPreference;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import edu.jhuapl.dorset.Response;
+import edu.jhuapl.dorset.ResponseStatus;
+import edu.jhuapl.dorset.http.HttpClient;
+
+public class StockAgent extends AbstractAgent {
+
+    private String baseurl = "https://www.quandl.com/api/v3/datasets/WIKI/";
+    private String apiKey;
+    private HttpClient client;
+    private TreeMap<String, String> stockSymbolMap;
+    private static final int DAYS_IN_A_MONTH = 30;
+    private static final Logger logger = LoggerFactory.getLogger(StockAgent.class);
+   
+    private static final String SUMMARY =
+            "Stock ticker that gives historical stock information for NASDAQ and NYSE companies";
+
+    private static final CellProcessor[] processors = new CellProcessor[] {
+            new NotNull(), new NotNull(), new NotNull(), new Optional(),
+            new Optional(), new Optional(), new NotNull(), new Optional() };
+
+    /**
+     * Stock agent
+     *
+     * @param client An http client object
+     * @param apiKey A Quandl API key
+     */
+    public StockAgent(HttpClient client, String apiKey) {
+        this.client = client;
+        this.apiKey = apiKey;
+
+        ICsvBeanReader csvBeanReader = null;
+
+        InputStream nasdaqCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NASDAQ_Companies.csv");
+
+        this.stockSymbolMap = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nasdaqCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+                this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NASDAQ_Companies.csv");
```

Comment:
upgrade this message to a higher level (fatal? error?) This means the stock agent won't work.